mempool/mempool.go

```diff
@@ -110,6 +115,29 @@ func (mem *Mempool) SetLogger(l log.Logger) {
 	mem.logger = l
 }
 
+func (mem *Mempool) OnStop() {
+	mem.BaseService.OnStop()
+	mem.Stop()
+}
+
+// Stop closes the underlying WAL file and then discards it.
+// After Stop has been invoked, any further writes will not be relayed to disk.
+func (mem *Mempool) Stop() bool {
```

Comment:
https://github.com/tendermint/tmlibs/blob/master/common/service.go#L124

func body should go into onStop